Sources/MIDIKit/MIDINetwork.swift

```diff
@@ -0,0 +1,199 @@
+        
+        removeAllNetworkConnections()
+        
+        let networkSession = self.networkSession ?? MIDINetworkSession()
+        
+        if let connection = lastKnownConnection, useLastKnownConnection {
+            networkSession.addConnection(connection)
+            try? start()
+
+            handler?(connection)
+            return
+        }
+    
+        bonjourService.findService(BonjourService.Services.Apple_Midi, domain: BonjourService.LocalDomain) { services in
+            
+            if let service = services.first, let hostName = service.hostName {
+                
+                let connection = MIDINetworkConnection(host: .init(name: service.name, address: hostName, port: service.port))
+                networkSession.addConnection(connection)
+                try? self.start()
+
+                self.lastKnownConnection = connection
+                handler?(connection)
+            } else {
+                handler?(nil)
+            }
+        }
+    }
+    
+    public var networkSourceEndpoint: MIDIEndpoint? {
+        if let endpoint = networkSession?.sourceEndpoint() {
+            return MIDIEndpoint(endpoint)
+        }
+        return nil
+    }
+    
+    public var networkDestinationEndpoint: MIDIEndpoint? {
+        if let endpoint = networkSession?.destinationEndpoint() {
+            return MIDIEndpoint(endpoint)
+        }
+        return nil
+    }
+    
+    public func removeAllNetworkConnections() {
+        for connection in networkSession?.connections() ?? [] {
+            networkSession?.removeConnection(connection)
+        }
+    }
+    
+    public private(set) var lastKnownConnection: MIDINetworkConnection? {
+        get {
+            let ud = UserDefaults.standard
+            
+            if let name = ud.string(forKey: "MIDIKit-networkConnection-name"),
+               let hostName  = ud.string(forKey: "MIDIKit-networkConnection-host-name") {
+                
+                let port = ud.integer(forKey: "MIDIKit-networkConnection-host-port")
+                
+                return MIDINetworkConnection(host: .init(name: name, address: hostName, port: port))
+            }
+            
+            return nil
+        }
+        set {
+            let ud = UserDefaults.standard
+            if let connection = newValue {
+                ud.setValue(connection.host.name, forKey: "MIDIKit-networkConnection-name")
+                ud.setValue(connection.host.address, forKey: "MIDIKit-networkConnection-host-name")
+                ud.setValue(connection.host.port, forKey: "MIDIKit-networkConnection-host-port")
+            } else {
+                ud.removeObject(forKey: "MIDIKit-networkConnection-name")
+                ud.removeObject(forKey: "MIDIKit-networkConnection-host-name")
+                ud.removeObject(forKey: "MIDIKit-networkConnection-host-port")
```

Comment:
Strings like `"MIDIKit-networkConnection-name"`, `"MIDIKit-networkConnection-host-name"` and `"MIDIKit-networkConnection-host-port"` should at least be put into a constant, preferable be configurable.

Reply:
Revisiting the last known connection persistence approach